byzcoin/api.go

```diff
@@ -441,6 +441,30 @@ func (c *Client) StreamTransactions(handler func(StreamingResponse, error)) erro
 	}
 }
 
+func (c *Client) signerCounterDecoder(buf []byte, data interface{}) error {
+	err := protobuf.Decode(buf, data)
+	if err != nil {
+		return errors.New("couldn't decode the counters reply: " + err.Error())
+	}
+
+	reply, ok := data.(*GetSignerCountersResponse)
+	if !ok {
+		return errors.New("wrong type of response")
+	}
+
+	// This assumes the client is up-to-date with the latest block which
+	// is usually right as it is updated after each GetProof. The goal is
+	// to insure that we got the data from the latest block.
+	// Note: index 0 is checked for backwards compatibility.
+	if reply.Index != 0 && c.Latest != nil {
+		if uint64(c.Latest.Index) > reply.Index {
```

Comment:
Nothing to change here, just wondering

Reply:
It's actually possible for `reply.Index` to be higher than the currently know latest block index by the client as blocks might have been created since the last GetProof or similar.

If that happens, it should accept the counter because the client should not add transaction since.